### DISCUSSION TITLE
(aws-cdk-lib/aws-lambda): Lambda from custom ECR docker image doesn't support tag with undescores

### DISCUSSION THREAD

**Discussion by ShantanuKumar** [aws/aws-cdk#23131] [General]
### Describe the bug

As far as I see from docker [documentation](https://docs.docker.com/engine/reference/commandline/tag/) that underscores are allowed

>  A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters.

On the other hand when using docker images with underscores in tags, it doesn't work. 

### Expected Behavior

Lambda should work with docker images' tags which contain underscores as per docker documentation

### Current Behavior

```typescript
new lambda.DockerImageFunction(this, "lambda", {
      functionName: "function",
      code: lambda.DockerImageCode.fromEcr(repo, {
        tagOrDigest: "custom_name_latest",
      }),
   },

});

```

I get this error
```
failed: Error: The stack named LambdaStack failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Resource handler returned message: "Source image <account_id>.dkr.ecr.<region>.amazonaws.com/<repo_name>:custom_name_latest" is not valid.
```

I checked this works when I replace `tagOrDigest` with just latest.

### Reproduction Steps

```typescript
import * as lambda from "aws-cdk-lib/aws-lambda";
new lambda.DockerImageFunction(this, "lambda", {
      functionName: "function",
      code: lambda.DockerImageCode.fromEcr(repo, {
        tagOrDigest: "custom_name_latest",
      }),
   },
});
```

### CDK CLI Version

2.31.0

### Node.js Version

16.13.0

### OS

MacOS and Linux

### Language

Typescript

**Comment by ShantanuKumar** [comment]
It seems like it's just not underscores, but also with dash i.e. `custom-image-latest` as tag doesn't work.

**Comment by peterwoodworth** [comment]
I'm not reproducing this @ShantanuKumar, there may be some other error going on for you like referring to a repo in the wrong region/account. The CloudFormation template generated looks perfectly fine too.

I'm moving this to discussions, that will be a better place for you to receive support on this topic 🙂

**Reply by ShantanuKumar** [reply]
Hey I have been swamped with work. That's why for now I made changes to just use `latest` tag without debugging further if anything else might be the issue. That's actually working with the same exact CDK env and any other needed settings.
